__tests__/app.test.js

```diff
@@ -206,87 +205,94 @@ describe('/api/articles/:article_id/comments', () => {
             expect(body.msg).toBe('Bad request');
           });
     });
-    test('GET: 204 returns an empty array when a valid article is passed that does not have any associated comments', () => {
-      return request(app)
-      .get('/api/articles/2/comments')
-      .expect(204)
-      .then(({body}) => {
-        expect(body).toEqual({});
-      });
   });
-    test('POST: 201 inserts new comment and returns comment to client', () => {
-        const newComment = {
-            username: 'butter_bridge',
-            body: 'new comment'
-        };
+    describe('POST', () => {
+      test('GET: 204 returns an empty array when a valid article is passed that does not have any associated comments', () => {
         return request(app)
-        .post('/api/articles/2/comments')
-        .send(newComment)
-        .expect(201)
-        .then(({body: {comment}}) => {
-          expect(comment).toMatchObject({
-            comment_id: 19,
-            author: 'butter_bridge',
-            votes: 0,
-            article_id: 2,
-            body: 'new comment',
-            created_at: expect.any(String)
-          });
+        .get('/api/articles/2/comments')
+        .expect(204)
+        .then(({body}) => {
+          expect(body).toEqual({});
         });
-    });
-    test("POST 400: responds with appropriate status and error message when request has invalid content", () => {
-        return request(app)
-          .post("/api/articles/5/comments")
-          .send({
-            username: "Simon",
-            body: 12789041
-          })
-          .expect(400)
-          .then(({ body: { msg } }) => {
-            expect(msg).toBe("Bad request");
-          });
       });
-      test("POST 404: sends an appropriate status and error message when given a valid but non-existent id", () => {
-        return request(app)
-          .post("/api/articles/999/comments")
-          .send({
-            username: "Mashca",
-            body: "New article is great!"
-          })
-          .expect(404)
-          .then(({ body: { msg } }) => {
-            expect(msg).toBe("No article found for article_id: 999");
-          });
-      });
-      test("POST 400: sends an appropriate status and error message when given an invalid id", () => {
-        return request(app)
-          .get("/api/articles/not-an-article/comments")
-          .send({
-            username: "Mashca",
-            body: "New article is great!"
-          })
-          .expect(400)
-          .then(({ body: { msg } }) => {
-            expect(msg).toBe("Bad request");
+      test('POST: 201 inserts new comment and returns comment to client', () => {
```

Comment:
Comments should only be able to post with a valid user from the user database. Since the username is a foreign key on the comments table, a non existent user should cause a psql error to be kicked out.

Lets write a test for this and make sure that, when everything else is OK but we give a non-existent user, a 404 error gets sent back.

Reply:
Ok! Will add